BorrowBookUI.java

```diff
@@ -0,0 +1,128 @@
+import java.util.Scanner;
+
+
+public class BorrowBookUI {
+	
+	public static enum UIState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+
+	private BorrowBookControl control;
+	private Scanner input;
+	private UIState state;
+
+	
+	public BorrowBookUI(BorrowBookControl control) {
+		this.control = control;
+		input = new Scanner(System.in);
```

Comment:
DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier